lib/sinon/spy.js

```diff
@@ -362,97 +198,40 @@ var spyApi = {
-delegateToCalls("alwaysCalledWithMatch", false, "calledWithMatch");
-delegateToCalls("calledWithExactly", true);
-delegateToCalls("calledOnceWithExactly", true, "calledWithExactly", false, undefined, 1);
-delegateToCalls("alwaysCalledWithExactly", false, "calledWithExactly");
-delegateToCalls("neverCalledWith", false, "notCalledWith", false, function() {
-    return true;
-});
-delegateToCalls("neverCalledWithMatch", false, "notCalledWithMatch", false, function() {
-    return true;
-});
-delegateToCalls("threw", true);
-delegateToCalls("alwaysThrew", false, "threw");
-delegateToCalls("returned", true);
-delegateToCalls("alwaysReturned", false, "returned");
-delegateToCalls("calledWithNew", true);
-delegateToCalls("alwaysCalledWithNew", false, "calledWithNew");
 /* eslint-disable local-rules/no-prototype-methods */
-delegateToCalls("callArg", false, "callArgWith", true, function() {
+var delegateToCalls = callUtil.delegateToCalls;
+delegateToCalls(spyApi, "callArg", false, "callArgWith", true, function() {
     throw new Error(this.toString() + " cannot call arg since it was not yet invoked.");
 });
 spyApi.callArgWith = spyApi.callArg;
-delegateToCalls("callArgOn", false, "callArgOnWith", true, function() {
+delegateToCalls(spyApi, "callArgOn", false, "callArgOnWith", true, function() {
     throw new Error(this.toString() + " cannot call arg since it was not yet invoked.");
 });
 spyApi.callArgOnWith = spyApi.callArgOn;
-delegateToCalls("throwArg", false, "throwArg", false, function() {
+delegateToCalls(spyApi, "throwArg", false, "throwArg", false, function() {
     throw new Error(this.toString() + " cannot throw arg since it was not yet invoked.");
 });
-delegateToCalls("yield", false, "yield", true, function() {
+delegateToCalls(spyApi, "yield", false, "yield", true, function() {
     throw new Error(this.toString() + " cannot yield since it was not yet invoked.");
 });
 // "invokeCallback" is an alias for "yield" since "yield" is invalid in strict mode.
 spyApi.invokeCallback = spyApi.yield;
-delegateToCalls("yieldOn", false, "yieldOn", true, function() {
+delegateToCalls(spyApi, "yieldOn", false, "yieldOn", true, function() {
     throw new Error(this.toString() + " cannot yield since it was not yet invoked.");
 });
-delegateToCalls("yieldTo", false, "yieldTo", true, function(property) {
+delegateToCalls(spyApi, "yieldTo", false, "yieldTo", true, function(property) {
     throw new Error(
         this.toString() + " cannot yield to '" + valueToString(property) + "' since it was not yet invoked."
     );
 });
-delegateToCalls("yieldToOn", false, "yieldToOn", true, function(property) {
+delegateToCalls(spyApi, "yieldToOn", false, "yieldToOn", true, function(property) {
     throw new Error(
         this.toString() + " cannot yield to '" + valueToString(property) + "' since it was not yet invoked."
     );
 });
 /* eslint-enable local-rules/no-prototype-methods */
 
+extend(spy, fakeApi);
```

Comment:
We're adding all the functions to the exported API here just for `stub.js` to have access to it. Now that the `fakeAPI` was moved into `proxy.js` and is applied there to each proxy instance, there's no need to do this anymore.